Clarifications to chunk format document.

nmoinvaz opened the pull request:
I have made some modifications for improved clarity. I am hoping I am understanding the format correctly.

FrancescAlted commented on README_CHUNK_FORMAT.rst:
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
     (``uint32``) Compressed size of the buffer (including this header).
 
+:filter_codes:
+    (``uint8x6``) Filter code.
```
uint8x6 -> uint8

FrancescAlted commented on README_CHUNK_FORMAT.rst:
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
+
+:filter_meta:
+
+    (``uint8x6``) Filter metadata.
```
ditto

FrancescAlted commented on README_CHUNK_FORMAT.rst:
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
+**Block starts**
 
-At the beginning of the blocks section, there come a list of `int32_t bstarts` to indicate where the different encoded blocks starts (counting from the end of this `bstarts` section)::
+The *block starts* section contains a list of offsets `int32_t bstarts` that indicate where each block starts in the
```
For consistency with the rest of the document, I think it is better int32_t -> int32

FrancescAlted commented on README_CHUNK_FORMAT.rst:
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
+*Only for C-Blosc2*
+
+Dictionaries are small datasets that are known to be repeated a lot and can help to compress data in blocks better.
+The dictionary section contains the size of the dictionary `int32_t dsize` followed by the dictionary data::
```
int32_t -> int32

FrancescAlted commented on README_CHUNK_FORMAT.rst:
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
-*Note*: all the integers are stored in little endian.
\ No newline at end of file
+    - When the most significant bit is *not* set, `csize` represents the size of the compressed
+      data stream that follows. (supported by C-Blosc1)
```
"supported by" -> "as in".  This is the only meaning for `csize` in C-Blosc1, so I think this reads better this way?

FrancescAlted commented on README_CHUNK_FORMAT.rst:
```diff
@@ -103,40 +92,108 @@ All entries are little endian.
+    - When the most significant bit is *not* set, `csize` represents the size of the compressed
+      data stream that follows. (supported by C-Blosc1)
+    - When the most significant bit is set, the lowest significant byte of `csize` is used to fill the entire
+      block.
```
Hmmm, you removed the examples.  Why so?

nmoinvaz commented:
My idea with the compression stream terminology comes from zlib/deflate. As in each time the data goes through the compression codec it outputs a compressed stream of data. With deflate, it processes the data, creating blocks with literals/matches, and then at the end of the stream it finalizes it with a last block.

nmoinvaz commented:
@FrancescAlted I have made the changes you have suggested. I also added to the _Compressed Data Streams_ section explaining what compressed data stream is.

FrancescAlted commented on README_CHUNK_FORMAT.rst:
```diff
@@ -103,40 +92,112 @@ All entries are little endian.
+If bit 4 of the `flags` header field is *not* set, each block of is stored in a single data stream::
+
+    +========+========+
+    | csize0 | cdata0 |
```
Why not use stream0 instead of cdata0?  It looks more consistent to me.

FrancescAlted commented on README_CHUNK_FORMAT.rst:
```diff
@@ -103,40 +92,112 @@ All entries are little endian.
 
     +========+========+========+========+========+========+========+
-    | csize0 | split0 | csize1 | split1 |   ...  | csizeN | splitN |
+    | csize0 | cdata0 | csize1 | cdata1 |   ...  | csizeN | cdataN |
```
Ditto

nmoinvaz commented:
I am taking a look at this again. What do you think of _compressed data blocks_ or _compressed data chunks_, instead of _compressed data streams_?

nmoinvaz commented:
Actually it looks like you were using _compressed blocks_ in your document previously so maybe that makes the most sense.

nmoinvaz commented:
@FrancescAlted I have made a few more modifications and renamed the sections to "streams" like you suggested, but also added another figure for `csize/cdata`.

FrancescAlted commented:
Thanks!

FrancescAlted commented:
Now we would need to update the [C-Blosc docs](https://github.com/Blosc/c-blosc/blob/master/README_CHUNK_FORMAT.rst) accordingly.

Also, there is reference inside the `blosc.c` and `blosc2.c` to splits.  That should be changed into streams.
